From 18a71132e9cd8b948602738b583333626692c6f3 Mon Sep 17 00:00:00 2001
From: Glenn Washburn <development@efficientek.com>
Date: Fri, 17 Sep 2021 23:03:59 +0000
Subject: [PATCH 046/261] tests: Only test MINIX3 volumes of 1 KiB block size

Apparently there used to be a -B option for mkfs.minix to create a volume
with a specified block size. This version is hard to come by and does not
appear to be available in Debian distributions. So, remove support for
testing a variety of blocks sizes for MINIX3. This allows the MINIX tests
to run because they were being skipped due to not finding a mkfs.minix with
the -B option.

Signed-off-by: Glenn Washburn <development@efficientek.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 tests/minixfs_test.in        | 5 -----
 tests/util/grub-fs-tester.in | 4 ++--
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/tests/minixfs_test.in b/tests/minixfs_test.in
index 3b16a4de0..437d92df6 100644
--- a/tests/minixfs_test.in
+++ b/tests/minixfs_test.in
@@ -25,11 +25,6 @@ if ! mkfs.minix -h | grep -- -3 > /dev/null; then
    exit 77
 fi
 
-if ! mkfs.minix -h | grep -- -B > /dev/null; then
-    echo "mkfs.minix doesn't support variable block size; cannot test minix*fs."
-   exit 77
-fi
-
 "@builddir@/grub-fs-tester" minix
 "@builddir@/grub-fs-tester" minix2
 "@builddir@/grub-fs-tester" minix3
diff --git a/tests/util/grub-fs-tester.in b/tests/util/grub-fs-tester.in
index 0fd2ecf45..099228c66 100644
--- a/tests/util/grub-fs-tester.in
+++ b/tests/util/grub-fs-tester.in
@@ -201,7 +201,7 @@ for LOGSECSIZE in $(range "$MINLOGSECSIZE" "$MAXLOGSECSIZE" 1); do
 		# OS LIMITATION: Linux rejects non-power-of-two blocks.
 		# OS LIMITATION: Linux rejects > 4096.
 	    MINBLKSIZE=1024
-	    MAXBLKSIZE=4096;;
+	    MAXBLKSIZE=1024;;
     esac
     if test "$BLKSTEP" -eq 0; then
 	blksizes="$(powrange "$MINBLKSIZE" "$MAXBLKSIZE")"
@@ -697,7 +697,7 @@ for LOGSECSIZE in $(range "$MINLOGSECSIZE" "$MAXLOGSECSIZE" 1); do
 		    "mkfs.minix" -v "${MOUNTDEVICE}"
 		    MOUNTFS="minix";;
 		x"minix3")
-		    "mkfs.minix" -B $BLKSIZE -3 "${MOUNTDEVICE}"
+		    "mkfs.minix" -3 "${MOUNTDEVICE}"
 		    MOUNTFS="minix";;
 		x"ntfs"*)
 		    "mkfs.ntfs" -s "$SECSIZE" -c "$BLKSIZE" -L "$FSLABEL" -Q -q "${MOUNTDEVICE}"
-- 
2.36.1

